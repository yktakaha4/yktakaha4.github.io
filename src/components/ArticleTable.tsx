import dayjs from "dayjs";
import {FC} from "react";

export interface ArticleTableRow {
    title: string
    url: string
    likes: number
    publishedAt: Date
}

export interface ArticleTableProps {
    rows: Array<ArticleTableRow>
}

export const ArticleTable: FC<ArticleTableProps> = ({ rows }) => {
    return (
        <table>
            <thead>
            <tr>
                <th style={{ textAlign: 'center' }}>公開日</th>
                <th>記事名</th>
            </tr>
            </thead>
            <tbody>
            {
                rows.map(row => (
                    <tr>
                        <td>{dayjs(row.publishedAt).format('YYYY/M/D')}</td>
                        <td><a href={row.url} target="_blank" rel="noopener noreferrer">{row.title}</a></td>
                    </tr>
                ))
            }
            </tbody>
        </table>
    )
}
