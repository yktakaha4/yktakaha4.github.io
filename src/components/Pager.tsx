import React, { FC } from 'react';

export interface PagerProps {
    page: number
    perPage: number
    total: number
    size: number
    onChange?: (args: { page: number }) => void
}

export const Pager: FC<PagerProps> = ({ page, perPage, total, size, onChange }) => {
    const minSize = 3
    if (size < minSize) {
        throw new Error(`size must be greater than or equal to ${minSize}`)
    }

    const totalPages = Math.ceil(total / perPage);
    const currentPage = Math.min(Math.max(1, page), totalPages);
    const startPage = Math.min(Math.max(1, currentPage - Math.floor(size / 2)), Math.max(totalPages - size + 1, 1));
    const pagesLength = Math.max(1, Math.min(size, totalPages))
    const pages = Array.from({ length: pagesLength }).map((_, i) => startPage + i);

    const handleClick = (p: number) => {
        return (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
            e.preventDefault();
            if (onChange) {
                onChange({ page: p })
            }
        }
    }

    return (
        <ul className="pagination" style={{margin: 0}}>
            <li className="pagination__item">
                <a className="pagination__link" href="#" onClick={handleClick(1)}>&laquo;</a>
            </li>
            {pages.length > 1 && pages[0] !== 1 && (
                <li className="pagination__item">
                    <span>…</span>
                </li>
            )}
            {pages.map(p => (
                <li key={p} className={`pagination__item ${p === currentPage ? 'pagination__item--active' : ''}`}>
                    <a className="pagination__link" href="#" onClick={handleClick(p)}>{p}</a>
                </li>
            ))}
            {pages.length > 1 && pages[pages.length - 1] !== totalPages && (
                <li className="pagination__item">
                    <span>…</span>
                </li>
            )}
            <li className="pagination__item">
                <a className="pagination__link" href="#" onClick={handleClick(totalPages)}>&raquo;</a>
            </li>
        </ul>
    );
}
