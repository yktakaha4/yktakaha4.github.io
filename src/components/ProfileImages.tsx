import { FC } from 'react';
import { Avatar } from '@/components/ui/Avatar';

export const ProfileImages: FC = () => {
  const src = [
    //
    require('@site/static/img/icon.jpg').default,
    require('@site/static/img/yktakaha4.jpg').default,
  ];
  return <Avatar src={src} />;
};
