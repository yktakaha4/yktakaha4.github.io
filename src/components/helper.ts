import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
export type CustomFields = {
  buildAt: string;
  commitHash: string;
  isDevelopment: boolean;
  isProduction: boolean;
  withEmbeddedContent: boolean;
};

export const getCustomFields = (): CustomFields => {
  const {
    siteConfig: { customFields },
  } = useDocusaurusContext();
  if (customFields) {
    return customFields as CustomFields;
  }
  throw new Error('customFields is not defined');
};
