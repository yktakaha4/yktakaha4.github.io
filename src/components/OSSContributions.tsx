import {FC, useMemo, useState} from "react";
import {OSSContribution} from "@/services/ossContributions";
import {createSearchRegexp, createSearchText, Search} from "@/components/ui/Search";
import dayjs from "dayjs";
import {Table, TableHeaders, TableRow} from "@/components/ui/Table";
import {getSliceIndex, Pager} from "@/components/ui/Pager";
import {getOSSContributionKindName, pagerPerPage, pagerSize} from "@/constants";

export type OSSContributionsProps = {
    contributions: Array<OSSContribution>
}

export const filterOSSContribution = (regexp: RegExp, contribution: OSSContribution) => {
    return regexp.test(createSearchText(
        contribution.title,
        dayjs(contribution.mergedAt).format('YYYY/M/D'),
        getOSSContributionKindName(contribution.kind),
        ...contribution.repository.languages,
        contribution.repository.name,
    ))
}

export const OSSContributions: FC<OSSContributionsProps> = ({ contributions }) => {
    const headers: TableHeaders = [{
        label: '日付',
        align: 'center',
        width: '10%',
    }, {
        label: 'タイトル',
        align: 'left',
        width: '40%',
    }, {
        label: 'リポジトリ',
        align: 'left',
        width: '40%',
    }, {
        label: 'タグ',
        align: 'left',
        width: '10%',
    }]

    const [query, setQuery] = useState('')
    const [page, setPage] = useState(1)

    const regexp = useMemo(() => createSearchRegexp(query), [query])

    const filtered = contributions.filter(contribution => filterOSSContribution(regexp, contribution))
    const rows = filtered.length > 0 ? filtered.map(({ mergedAt, title, url, repository, kind }): TableRow => {
        return [{
            type: 'date',
            value: mergedAt,
        }, {
            type: 'string',
            value: title,
            link: {
                type: 'external',
                href: url,
            },
        }, {
            type: 'string',
            value: repository.name,
            link: {
                type: 'external',
                href: repository.url,
            },
        }, {
            type: 'tags',
            values: [{
                icon: '📝',
                value: getOSSContributionKindName(kind),
                color: 'primary',
            }],
        }]
    }).slice(...getSliceIndex(page, pagerPerPage)) : []

    const handleChangeQuery = (newQuery: string) => {
        setQuery(newQuery)
        setPage(1)
    }
    const handleChangePage = (newPage: number) => {
        setPage(newPage)
    }

    return (<>
        <div style={{marginBottom: '0.2rem'}}>
            <Search
                query={query}
                totalCount={filtered.length}
                onChange={({ query }) => handleChangeQuery(query)}
                children={<Pager
                    page={page}
                    perPage={pagerPerPage}
                    size={pagerSize}
                    total={filtered.length}
                    onChange={({ page }) => handleChangePage(page)}
                />}
            />
        </div>
        <Table headers={headers} rows={rows} />
    </>)
}
