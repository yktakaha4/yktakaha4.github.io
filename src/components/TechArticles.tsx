import {FC, useMemo, useState} from "react";
import {createSearchRegexp, createSearchText, Search} from "@/components/ui/Search";
import {getSliceIndex, Pager} from "@/components/ui/Pager";
import {Table, TableHeaders, TableRow, TableRows} from "@/components/ui/Table";
import {pagerPerPage, pagerSize, TechArticlePublisher, getTechArticlePublisherName} from "@/constants";
import dayjs from "dayjs";

export type TechArticle = {
    publishedAt: Date
    title: string
    url: string
    likes: number
    publisher: TechArticlePublisher
    tags: Array<string>
}

export type TechArticlesProps = {
    articles: Array<TechArticle>
}

export const filterTechArticle = (regexp: RegExp, article: TechArticle) => {
    return regexp.test(createSearchText(
        article.title,
        ...article.tags,
        getTechArticlePublisherName(article.publisher),
        dayjs(article.publishedAt).format('YYYY/M/D'),
    ))
}

export const TechArticles: FC<TechArticlesProps> = ({ articles }) => {
    const headers: TableHeaders = [{
        label: 'å…¬é–‹æ—¥',
        align: 'center',
        width: '10%',
    }, {
        label: 'ã‚¿ã‚¤ãƒˆãƒ«',
        align: 'left',
        width: '70%',
    }, {
        label: 'ã‚¿ã‚°',
        align: 'left',
        width: '20%',
    }]

    const [query, setQuery] = useState('')
    const [page, setPage] = useState(1)

    const regexp = useMemo(() => createSearchRegexp(query), [query])

    const filteredArticles = articles.filter(article => filterTechArticle(regexp, article))
    const rows: TableRows = filteredArticles.map(({ publishedAt, title, url, likes, tags, publisher}): TableRow => {
        return [{
            type: 'date',
            value: publishedAt,
            format: 'YYYY/M/D',
        }, {
            type: 'string',
            value: title,
            link: {
                type: 'external',
                href: url,
            },
        }, {
            type: 'tags',
            values: [{
                icon: 'ðŸ“°',
                value: getTechArticlePublisherName(publisher),
                color: 'primary',
            }, {
                icon: 'â™¥',
                value: likes.toLocaleString(),
                color: 'danger',
            }, ...tags.map((tag) => ({ value: tag }))],
        }]
    }).slice(...getSliceIndex(page, pagerPerPage))

    const handleChangeQuery = (newQuery: string) => {
        setQuery(newQuery)
        setPage(1)
    }
    const handleChangePage = (newPage: number) => {
        setPage(newPage)
    }

    return (<>
        <div style={{marginBottom: '0.2rem'}}>
            <Search
                query={query}
                onChange={({ query }) => handleChangeQuery(query)}
                children={<Pager
                    page={page}
                    perPage={pagerPerPage}
                    size={pagerSize}
                    total={filteredArticles.length}
                    onChange={({ page }) => handleChangePage(page)}
                />}
            />
        </div>
        <Table headers={headers} rows={rows} />
    </>)
}
