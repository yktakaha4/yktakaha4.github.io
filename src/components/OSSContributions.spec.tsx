import { render } from '@testing-library/react';
import { OSSContributions } from '@/components/OSSContributions';

describe('OSSContributions', () => {
  describe('スナップショットテスト', () => {
    test('OSS活動が描画される', () => {
      const { container } = render(<OSSContributions />);
      expect(container).toMatchSnapshot();
    });
  });
});
