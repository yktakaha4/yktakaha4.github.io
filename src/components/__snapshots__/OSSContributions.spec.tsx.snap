// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OSSContributions „Çπ„Éä„ÉÉ„Éó„Ç∑„Éß„ÉÉ„Éà„ÉÜ„Çπ„Éà OSSÊ¥ªÂãï„ÅåÊèèÁîª„Åï„Çå„Çã 1`] = `
<div>
  <div
    style="margin-bottom: 0.2rem;"
  >
    <nav
      class="navbar"
      style="margin: 0px; padding: 0px; background-color: transparent; box-shadow: none; height: inherit;"
    >
      <div
        class="navbar__inner"
      >
        <div
          class="navbar__items"
        >
          <div
            class="navbar__search"
          >
            <input
              class="navbar__search-input"
              maxlength="100"
              type="text"
              value=""
            />
          </div>
          <div
            class="navbar__item"
          >
            31
             ‰ª∂
          </div>
        </div>
        <div
          class="navbar__items navbar__items--right"
        >
          <div
            class="navbar__item"
          >
            <ul
              class="pagination"
              style="margin: 0px;"
            >
              <li
                class="pagination__item"
              >
                <a
                  class="pagination__link"
                  data-mock="LinkMock"
                  href="#"
                  to="#"
                >
                  ¬´
                </a>
              </li>
              <li
                class="pagination__item pagination__item--active"
              >
                <a
                  class="pagination__link"
                  data-mock="LinkMock"
                  href="#"
                  to="#"
                >
                  1
                </a>
              </li>
              <li
                class="pagination__item "
              >
                <a
                  class="pagination__link"
                  data-mock="LinkMock"
                  href="#"
                  to="#"
                >
                  2
                </a>
              </li>
              <li
                class="pagination__item "
              >
                <a
                  class="pagination__link"
                  data-mock="LinkMock"
                  href="#"
                  to="#"
                >
                  3
                </a>
              </li>
              <li
                class="pagination__item"
              >
                <span>
                  ‚Ä¶
                </span>
              </li>
              <li
                class="pagination__item"
              >
                <a
                  class="pagination__link"
                  data-mock="LinkMock"
                  href="#"
                  to="#"
                >
                  ¬ª
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </nav>
  </div>
  <table
    style="margin: 0px;"
  >
    <thead>
      <tr>
        <th
          style="text-align: center; width: 10%;"
        >
          Êó•‰ªò
        </th>
        <th
          style="text-align: left; width: 40%;"
        >
          „Çø„Ç§„Éà„É´
        </th>
        <th
          style="text-align: left; width: 40%;"
        >
          „É™„Éù„Ç∏„Éà„É™
        </th>
        <th
          style="text-align: left; width: 10%;"
        >
          „Çø„Ç∞
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          2023/11/9
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/hashicorp/terraform-provider-aws/pull/34312"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/hashicorp/terraform-provider-aws/pull/34312"
          >
            docs: fix resource id
          </a>
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/hashicorp/terraform-provider-aws"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/hashicorp/terraform-provider-aws"
          >
            hashicorp/terraform-provider-aws
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <span
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              üìù 
              PR
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2023/10/17
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/aws/aws-eks-best-practices/pull/398"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/aws/aws-eks-best-practices/pull/398"
          >
            fix broken link
          </a>
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/aws/aws-eks-best-practices"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/aws/aws-eks-best-practices"
          >
            aws/aws-eks-best-practices
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <span
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              üìù 
              PR
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2023/9/14
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/helmfile/helmfile/pull/1019"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/helmfile/helmfile/pull/1019"
          >
            Add diffArgs to helmDefaults
          </a>
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/helmfile/helmfile"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/helmfile/helmfile"
          >
            helmfile/helmfile
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <span
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              üìù 
              PR
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2023/6/17
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/hashicorp/terraform-provider-aws/pull/27604"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/hashicorp/terraform-provider-aws/pull/27604"
          >
            Add import of \`aws_ses_active_receipt_rule_set\` resource
          </a>
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/hashicorp/terraform-provider-aws"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/hashicorp/terraform-provider-aws"
          >
            hashicorp/terraform-provider-aws
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <span
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              üìù 
              PR
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2023/5/7
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/ebitengine/ebitengine.org/pull/17"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/ebitengine/ebitengine.org/pull/17"
          >
            add about go run in wsl
          </a>
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/ebitengine/ebitengine.org"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/ebitengine/ebitengine.org"
          >
            ebitengine/ebitengine.org
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <span
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              üìù 
              PR
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2023/3/20
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/cloudflare/terraform-provider-cloudflare/pull/2298"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/cloudflare/terraform-provider-cloudflare/pull/2298"
          >
            add import \`cloudflare_tunnel_config\`
          </a>
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/cloudflare/terraform-provider-cloudflare"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/cloudflare/terraform-provider-cloudflare"
          >
            cloudflare/terraform-provider-cloudflare
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <span
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              üìù 
              PR
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2023/2/23
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/FairwindsOps/pluto/pull/457"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/FairwindsOps/pluto/pull/457"
          >
            add: set replacement-available-in to existing version.
          </a>
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/FairwindsOps/pluto"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/FairwindsOps/pluto"
          >
            FairwindsOps/pluto
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <span
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              üìù 
              PR
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2023/2/22
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/FairwindsOps/pluto/pull/452"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/FairwindsOps/pluto/pull/452"
          >
            add replacement-available-in
          </a>
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/FairwindsOps/pluto"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/FairwindsOps/pluto"
          >
            FairwindsOps/pluto
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <span
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              üìù 
              PR
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2023/2/4
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/FairwindsOps/pluto/pull/448"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/FairwindsOps/pluto/pull/448"
          >
            fix: add removed-in to storage.k8s.io/v1beta1.CSIStorageCapacity
          </a>
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/FairwindsOps/pluto"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/FairwindsOps/pluto"
          >
            FairwindsOps/pluto
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <span
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              üìù 
              PR
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          2023/2/2
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/argoproj/argo-workflows/pull/10421"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/argoproj/argo-workflows/pull/10421"
          >
            fix: add message when parse of private key fails due to existing sso secret. Fixes #10420
          </a>
        </td>
        <td>
          <a
            data-mock="LinkMock"
            href="https://github.com/argoproj/argo-workflows"
            rel="noopener noreferrer"
            target="_blank"
            to="https://github.com/argoproj/argo-workflows"
          >
            argoproj/argo-workflows
          </a>
        </td>
        <td>
          <div
            style="display: flex; gap: 0.2rem; flex-wrap: wrap;"
          >
            <span
              class="badge badge--primary"
              style="white-space: nowrap;"
            >
              üìù 
              PR
            </span>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
